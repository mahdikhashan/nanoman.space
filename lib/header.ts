import { HeaderItem } from "@/types";

export const header: HeaderItem[] = [
  {
    name: 'Home',
    slug: '',
    isDisabled: true,
  },
  {
    name: 'About',
    slug: 'about',
    isDisabled: true
  },
  {
    name: 'Blog',
    slug: 'blog',
    isDisabled: true
  },
  {
    name: 'Projects',
    slug: 'projects',
    isDisabled: true,
    items: [
      {
        name: 'Hooks',
        slug: 'hooks',
        description:
          'Preview the hooks available for Client and Server Components',
      },
      {
        name: 'Client Context',
        slug: 'context',
        description:
          'Pass context between Client Components that cross Server/Client Component boundary',
      },
    ],
  },
  {
    name: 'Travel',
    slug: 'travels',
    isDisabled: true
  },
  {
    name: 'LinkedIn',
    slug: 'https://www.linkedin.com/in/mahdikhashan/',
    redirect: true,
    isDisabled: true
  },
  {
    name: 'Github',
    slug: 'https://github.com/mahdikhashan',
    redirect: true,
    isDisabled: true
  },
  {
    name: 'Twitter',
    slug: 'https://twitter.com/__nanoman__',
    redirect: true,
    isDisabled: true
  },
];
