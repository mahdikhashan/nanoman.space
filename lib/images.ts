import { PhotoImage } from "@/types";

export const images: PhotoImage[] = [
  { src: '/images/me_in_laft.jpeg', width: 400, height: 200, alt: "me" },
  { src: '/images/me_in_laft_w.jpeg', width: 400, height: 800, alt: "me" },
  { src: '/images/me_in_laft_t.jpeg', width: 200, height: 100, alt: "me" },
];
