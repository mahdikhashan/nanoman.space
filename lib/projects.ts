type Project = {
  id: number;
  name: string;
  icon: string;
  slug: string;
  github: string;
  website: string;
  description: string;
  skills: string[];
  featured: boolean;
  type: string;
};

const projects: Project[] = [
  {
    id: 1,
    name: '99 Emails',
    icon: '/logo.svg',
    slug: '99-minutes-emails',
    github: 'https://github.com/mahdikhashan/99-minutes-emails',
    website: 'https://99-minutes-emails.vercel.app/',
    description: 'A user friendly, safe and available temporary email service.',
    skills: [
      'react',
      'typescript',
      'redux-saga',
      'chakra-ui',
      'vite',
      'vitest',
      'cypress',
      'storybook'
    ],
    featured: true,
    type: 'Side Project'
  },
  {
    id: 2,
    name: 'Aparat Python API',
    icon: '/package.png',
    slug: 'aparat-python',
    github: 'https://github.com/mahdikhashan/aparat-python',
    website: 'https://pypi.org/project/aparat-python/',
    description: 'Aparat VOD Platform API Wrapper',
    skills: ['python', 'REST-API', 'unit-test'],
    featured: true,
    type: 'Python Package'
  },
  {
    id: 3,
    name: 'Vue 3 Click Outside',
    icon: '/package.png',
    slug: 'vue3-click-outside',
    github: 'https://github.com/mahdikhashan/vue3-click-outside',
    website: 'https://www.npmjs.com/package/@mahdikhashan/vue3-click-outside',
    description:
      'Directive for Vue 3 to run a method on clicking outside of the binded element',
    skills: ['javascript', 'vue3', 'vue-test-utils', 'jest'],
    featured: true,
    type: 'NPM Package'
  },
  {
    id: 4,
    name: 'Vercel CORS Proxy',
    icon: '/package.png',
    slug: 'vercel-express-proxy',
    github: 'https://github.com/mahdikhashan/vercel-express-proxy',
    website: 'https://github.com/mahdikhashan/vercel-express-proxy',
    description: 'ðŸªž Proxy server based on Express for Vercel platform',
    type: 'Template / CLI',
    featured: false,
    skills: ['javascript', 'node.js', 'express.js', 'vercel']
  },
  {
    id: 5,
    name: 'Vercel Express',
    icon: '/package.png',
    slug: 'vercel-express-template',
    github: 'https://github.com/mahdikhashan/vercel-express-template',
    website: 'https://github.com/mahdikhashan/vercel-express-template',
    description: 'ðŸ”¥ Express Server Template for Vercel',
    type: 'Template / CLI',
    featured: false,
    skills: ['javascript', 'node.js', 'express.js', 'vercel']
  },
  {
    id: 6,
    name: 'Veemoji',
    icon: '/package.png',
    slug: 'veemoji',
    github: 'https://github.com/mahdikhashan/veemoji',
    website: 'https://veemoji.surge.sh/',
    description: 'A dead simple emoji component library',
    type: 'NPM Package',
    featured: true,
    skills: ['javascript', 'vue', 'npm']
  }
];

export { projects, type Project };
