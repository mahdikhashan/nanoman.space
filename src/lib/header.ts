type Item = {
  name: string;
  slug: string;
  isDisabled?: boolean;
  redirect?: boolean;
  items?: {
    name: string;
    slug: string;
    description?: string;
    isDisabled?: boolean;
  }[];
};

export const header: Item[] = [
  {
    name: 'Home',
    slug: '/'
  },
  {
    name: 'About',
    slug: '/about'
  },
  {
    name: 'Projects',
    slug: '/projects',
    items: [
      {
        name: 'Hooks',
        slug: 'hooks',
        description:
          'Preview the hooks available for Client and Server Components',
      },
      {
        name: 'Client Context',
        slug: 'context',
        description:
          'Pass context between Client Components that cross Server/Client Component boundary',
      },
    ],
  },
  {
    name: 'LinkedIn',
    slug: 'https://www.linkedin.com/in/mahdikhashan/',
    redirect: true,
    isDisabled: true
  },
  {
    name: 'Github',
    slug: 'https://github.com/mahdikhashan',
    redirect: true,
    isDisabled: true
  },
  {
    name: 'Twitter',
    slug: 'twitter',
    redirect: true,
    isDisabled: true
  },
];
