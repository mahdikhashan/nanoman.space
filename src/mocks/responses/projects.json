[
  {
    "name": "Example Category",
    "slug": "example-category"
  },
  {
    "name": "LIFE",
    "slug": "life"
  }
]