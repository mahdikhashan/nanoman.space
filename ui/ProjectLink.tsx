import type { Project } from '@/lib/projects';
import Link from 'next/link';
import Image from 'next/image';

type Props = Project;

export default function ProjectLink({ name, description, website, icon }: Props) {
  return (
    <Link
      href={website}
      target="_blank"
      className="flex gap-5 space-y-1.5 rounded-lg border border-black/10 dark:border-white/10 px-4 py-3 hover:border-black/20 dark:hover:border-white/20 dark:bg-zinc-800 bg-white transition ease-in-out delay-50 hover:scale-[1.025]"
    >
      <img src={icon} alt={name} style={{ width: "50px", height: "50px" }} />
      <div>
        <div>{name}</div>
        <div className="line-clamp-3 text-sm text-zinc-400">{description}</div>
      </div>
    </Link>
  );
}
