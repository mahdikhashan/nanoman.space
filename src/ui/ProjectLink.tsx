import { Category } from '@/lib/types';
import Link from 'next/link';

type Props = Category;

export default function ProjectLink({ slug, name }: Props) {
  return (
    <Link
      href={`/`}
      key={slug}
      className="block space-y-1.5 rounded-lg border border-black/10 dark:border-white/10 px-4 py-3 hover:border-black/20 dark:hover:border-white/20 dark:bg-zinc-800 bg-white transition ease-in-out delay-50 hover:scale-[1.025]"
    >
      <div>{name}</div>
      <div className="line-clamp-3 text-sm text-zinc-400">{slug}</div>
    </Link>
  );
}
