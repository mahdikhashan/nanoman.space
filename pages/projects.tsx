import { InferGetStaticPropsType } from 'next';

import Container from '@/ui/Container';
import { Project, projects } from '@/lib/projects';
import ProjectLink from '@/ui/ProjectLink';

const prettyProjects = projects
  .map(function (project) {
    return {
      ['type']: project.type,
      ['projects']: projects.filter((p) => p.type == project.type)
    };
  })
  .reduce((acc, current) => {
    const x = acc.find((item) => item.type === current.type);
    if (!x) {
      return acc.concat([current]);
    } else {
      return acc;
    }
  }, []);

export default function ProjectsPage({
  projects
}: InferGetStaticPropsType<typeof getStaticProps>) {
  return (
    <Container>
      <div className="space-y-6 font-mono">
        <div className="space-y-8">
          <div className="mt-12 max-w-screen-md space-y-4 prose">
            <h1 className="dark:text-white">Projects</h1>
            <p className="dark:text-white">
              Things I've made over the years while trying to learn programming.
            </p>
          </div>

          <div className="flex-col space-y-2">
            {prettyProjects.map(({ type, projects }: { type: string, projects: Project[] }, i) => (
              <div key={i} className="flex flex-col gap-3 mt-6">
                <h3 className="text-lg font-bold text-gray-500 mt-3">{type}</h3>
                {projects.map((project, i) => (
                  <ProjectLink key={i} {...project} />
                ))}
              </div>
            ))}
            {!projects.length && <div>No Project found.</div>}
          </div>
        </div>
      </div>
    </Container>
  );
}

export async function getStaticProps() {
  return { props: { projects } };
}
