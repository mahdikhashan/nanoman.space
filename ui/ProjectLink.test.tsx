import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import ProjectLink from './ProjectLink';
import { Project } from '@/lib/projects';

const project: Project = {
  id: 1,
  name: '99 Emails',
  icon: '/logo.svg',
  slug: '99-minutes-emails',
  github: 'https://github.com/mahdikhashan/99-minutes-emails',
  website: 'https://99-minutes-emails.vercel.app/',
  description: 'A user friendly, safe and available temporary email service.',
  skills: [
    'react',
    'typescript',
    'redux-saga',
    'chakra-ui',
    'vite',
    'vitest',
    'cypress',
    'storybook'
  ],
  featured: true,
  type: 'sideproject'
};

describe('<ProjectLink />', () => {
  test('renders correctly', () => {
    const { container } = render(<ProjectLink {...project} />);
    expect(container).toMatchSnapshot();
  });
});
