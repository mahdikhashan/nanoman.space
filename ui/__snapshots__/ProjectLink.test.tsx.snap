// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectLink /> renders correctly 1`] = `
<div>
  <a
    class="flex gap-5 space-y-1.5 rounded-lg border border-black/10 dark:border-white/10 px-4 py-3 hover:border-black/20 dark:hover:border-white/20 dark:bg-zinc-800 bg-white transition ease-in-out delay-50 hover:scale-[1.025]"
    href="https://99-minutes-emails.vercel.app/"
    target="_blank"
  >
    <img
      alt="99 Emails"
      src="/logo.svg"
      style="width: 50px; height: 50px;"
    />
    <div>
      <div>
        99 Emails
      </div>
      <div
        class="line-clamp-3 text-sm text-zinc-400"
      >
        A user friendly, safe and available temporary email service.
      </div>
    </div>
  </a>
</div>
`;
